File: backend/src/types/task.ts
import type { ResourceDto } from './resource';

export type TaskStatus = 'pending' | 'running' | 'success' | 'failed';
export type StepStatus = 'pending' | 'success' | 'failed';
export type LogLevel = 'info' | 'success' | 'warning' | 'error';

export interface TransferTaskRecord {
  id: string;
  keyword: string;
  provider: string;
  resource_title: string;
  resource_payload: string;
  status: TaskStatus;
  transfer_status: StepStatus;
  strm_status: StepStatus;
  error_message: string | null;
  created_at: string;
  updated_at: string;
}

export interface TransferTaskDto {
  id: string;
  keyword: string;
  provider: string;
  resourceTitle: string;
  resource: ResourceDto;
  status: TaskStatus;
  transferStatus: StepStatus;
  strmStatus: StepStatus;
  errorMessage: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface TaskLogRecord {
  id: number;
  task_id: string;
  level: LogLevel;
  message: string;
  detail: string | null;
  created_at: string;
}

File: backend_py/app/models/task.py
from typing import Literal

from pydantic import BaseModel

from .resource import ResourceDto

StepStatus = Literal["pending", "success", "failed"]
TaskStatus = Literal["pending", "running", "success", "failed"]
LogLevel = Literal["info", "success", "error"]


class TaskDto(BaseModel):
    id: str
    keyword: str
    provider: str
    resourceTitle: str
    resource: ResourceDto
    status: TaskStatus
    transferStatus: StepStatus
    strmStatus: StepStatus
    errorMessage: str | None
    createdAt: str
    updatedAt: str


class TaskLogDto(BaseModel):
    id: int
    taskId: str
    level: LogLevel
    message: str
    createdAt: str
